notification/service/notification.go

```diff
@@ -14,65 +14,62 @@ import (
 	"github.com/fabric8-services/fabric8-auth/notification"
 	"github.com/fabric8-services/fabric8-auth/notification/client"
 	"github.com/fabric8-services/fabric8-auth/rest"
+	"github.com/fabric8-services/fabric8-auth/test/configuration"
 	"github.com/fabric8-services/fabric8-auth/token/signer"
+	goaclient "github.com/goadesign/goa/client"
 	"github.com/goadesign/goa/uuid"
+	"net/http"
 )
 
-// devNullNotificationService is the default dev service implementation. It does nothing.
-type devNullNotificationService struct{}
-
-func (s *devNullNotificationService) SendAsync(ctx context.Context, msg notification.Message) error {
-	return nil
-}
-
-func (s *devNullNotificationService) SendMessagesAsync(ctx context.Context, messages []notification.Message) error {
-	return nil
-}
-
 type notificationServiceImpl struct {
 	base.BaseService
 	config notification.Configuration
-	doer   rest.HttpDoer
 }
 
 // NewNotificationService creates a new service.
 func NewNotificationService(context servicecontext.ServiceContext, config notification.Configuration) service.NotificationService {
-	if config.GetNotificationServiceURL() == "" {
-		return &devNullNotificationService{}
-	}
 	return &notificationServiceImpl{
 		BaseService: base.NewBaseService(context),
 		config:      config,
-		doer:        rest.DefaultHttpDoer(),
 	}
 }
 
-// SendAsync creates a new goroutine and sends a message to fabric8-notification service
-func (s *notificationServiceImpl) SendAsync(ctx context.Context, msg notification.Message) error {
-	c, err := s.createClientWithContextSigner(ctx)
+// SendMessageAsync creates a new goroutine and sends a message to fabric8-notification service
+func (s *notificationServiceImpl) SendMessageAsync(ctx context.Context, msg notification.Message, options ...configuration.HTTPClientOption) (chan error, error) {
+	c, err := s.createClientWithContextSigner(ctx, options...)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	go s.send(ctx, c, msg)
+	errs := make(chan error, 1)
+	go func() {
+		defer close(errs)
+		if e := s.send(ctx, c, msg); e != nil {
+			errs <- e
+		}
+	}()
 
-	return nil
+	return errs, nil
 }
 
 // SendMessagesAsync creates a new goroutine and sends multiple messages to fabric8-notification service
-func (s *notificationServiceImpl) SendMessagesAsync(ctx context.Context, messages []notification.Message) error {
-	c, err := s.createClientWithContextSigner(ctx)
+func (s *notificationServiceImpl) SendMessagesAsync(ctx context.Context, messages []notification.Message, options ...configuration.HTTPClientOption) (chan error, error) {
```

Comment:
could you include a comment to explain the returned `chan error` and `error`: what they are used for, etc.